web/src/upload.js

```diff
@@ -411,6 +411,7 @@ export function setup_upload(config) {
         show_error_message(config, message, file.id);
         compose_ui.replace_syntax(get_translated_status(file), "", get_item("textarea", config));
         compose_ui.autosize_textarea(get_item("textarea", config));
+        uppy.removeFile(file.id);
```

Comment:
Looks like the correct fix! Let's add a comment explaining why we remove the file; you can refer to the comment right before we remove a file when an upload is successful.